mlflow/tracking/client.py

```diff
@@ -2119,6 +2119,31 @@ def set_experiment_tag(self, experiment_id: str, key: str, value: Any) -> None:
         """
         self._tracking_client.set_experiment_tag(experiment_id, key, value)
 
+    def delete_experiment_tag(self, experiment_id: str, key: str) -> None:
+        """
+        Delete a tag from the experiment with the specified ID.
+
+        Args:
+            experiment_id: String ID of the experiment.
+            key: Name of the tag to be deleted.
+
+        .. code-block:: python
+
+            from mlflow import MlflowClient
+
+            # Create an experiment and set its tag
+            client = MlflowClient()
+            experiment_id = client.create_experiment("Social Media NLP Experiments")
+            client.delete_experiment_tag(experiment_id, "nlp.framework")
+
+            # Fetch experiment metadata information
+            experiment = client.get_experiment(experiment_id)
+            print(f"Name: {experiment.name}")
+            print(f"Tags: {experiment.tags}")
+
```

Comment:
For this example, maybe we can set the tag first, then delete it
```suggestion
            from mlflow import MlflowClient

            # Create an experiment and set its tag
            client = MlflowClient()
            experiment_id = client.create_experiment("Social Media NLP Experiments")
            client.set_experiment_tag(experiment_id, "nlp.framework", "Spark NLP")

            # Fetch experiment metadata information, validate that tag is set
            experiment = client.get_experiment(experiment_id)
            print(f"Name: {experiment.name}")
            print(f"Tags: {experiment.tags}")

            client.delete_experiment_tag(experiment_id, "nlp.framework")

            # Fetch experiment metadata information, validate that tag is deleted
            experiment = client.get_experiment(experiment_id)
            print(f"Name: {experiment.name}")
            print(f"Tags: {experiment.tags}")
```

Reply:
Let's simplify the example. We don't need to show the experiment name at all. We should use `assert` to ensure that the tag has been deleted.